src/stubs/reactive_stubs_support.h

```diff
@@ -12,20 +12,42 @@ typedef enum
 {
     Ok                = 0x0,
     IllegalConnection = 0x1,
-    MalformedPayload  = 0x2
+    MalformedPayload  = 0x2,
+    InternalError     = 0x3
 } ResultCode;
 
+typedef struct Connection {
+    io_index    io_id;
+    conn_index  conn_id;
+    uint16_t    nonce;
+    IoKey       key;
+} Connection;
+
+// The size of the Connection struct is also hardcoded in linker.py. Hence,
+// we need to make sure that it does not change at compile time (e.g. due to
+// optimizations).
+// Besides, if the struct changes, we need to adjust this value here and in
+// linker.py (check the CONNECTION_STRUCT_SIZE global variable) as well.
+_Static_assert (sizeof(Connection) == 6 + SANCUS_KEY_SIZE,
+    "Size of Connection struct differs from the expected value");
+
 // These will be allocated by the linker
-extern IoKey __sm_io_keys[];
+extern Connection __sm_io_connections[];
 extern InputCallback __sm_input_callbacks[];
-extern uint16_t __sm_output_nonce;
 
-extern char __sm_num_connections;
-#define SM_NUM_CONNECTIONS (size_t)&__sm_num_connections
+extern char __sm_max_connections;
+#define SM_MAX_CONNECTIONS (size_t)&__sm_max_connections
```

Comment:
not sure what happens here: why is `__sm_max_connections` a char and not a uint16_t, and why do you take the address using `&`? The check `     if (__sm_num_connections == SM_MAX_CONNECTIONS) {` in sm_set_key will then compare an address vs an integer right??